vector/src/main/java/im/vector/app/features/location/LocationSharingServiceConnection.kt

```diff
@@ -49,30 +51,46 @@ class LocationSharingServiceConnection @Inject constructor(
         }
     }
 
-    fun unbind() {
-        callback = null
-    }
-
-    fun stopLiveLocationSharing(roomId: String) {
-        locationSharingService?.stopSharingLocation(roomId)
+    fun unbind(callback: Callback) {
+        removeCallback(callback)
     }
 
     override fun onServiceConnected(className: ComponentName, binder: IBinder) {
         locationSharingService = (binder as LocationSharingService.LocalBinder).getService().also {
             it.callback = this
         }
         isBound = true
-        callback?.onLocationServiceRunning()
+        onCallbackActionNoArg(Callback::onLocationServiceRunning)
     }
 
     override fun onServiceDisconnected(className: ComponentName) {
         isBound = false
         locationSharingService?.callback = null
         locationSharingService = null
-        callback?.onLocationServiceStopped()
+        onCallbackActionNoArg(Callback::onLocationServiceStopped)
     }
 
     override fun onServiceError(error: Throwable) {
-        callback?.onLocationServiceError(error)
+        forwardErrorToCallbacks(error)
+    }
+
+    @Synchronized
```

Comment:
do these need to be synchronised? Ideally we would only add/remove the callbacks from 1 thread (such as the main thread)